.github/workflows/workflow-validation.yml

```diff
@@ -0,0 +1,29 @@
+name: Workflow Validation Testing
+
+on:
+  pull_request:
+    branches:
+      - main
+      - develop
+      - 'feature/**'
+    paths:
+      - '.github/**.yml'
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - ready_for_review
+
+concurrency:
+  group: workflow-validation-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  visual-regression:
+    name: Run Workflow Validation
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    if: github.event.pull_request.draft == false
```

Comment:
Let's not add this as the job runs quite quickly and would be valuable for drafts as well. This will only run for PRs that change GHA workflow files so it's not something we need to worry about running too often.